.github/workflows/build-emsdk.yml

```diff
@@ -1,27 +1,78 @@
-name: build-emsdk
+name: Build WASM
 
-on: [push]
+# Run CI only on changes to main branch, or any PR to main. Do not run CI on 
+# any other branch. Also, skip any non-source changes from running on CI
+on:
+  push:
+    branches: main
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '.gitignore'
+      - '*.rst'
+      - '*.md'
+      - '.github/workflows/*.yml'
+      # re-include current file to not be excluded
+      - '!.github/workflows/build-emsdk.yml'
+
+  pull_request:
+    branches: main
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '.gitignore'
+      - '*.rst'
+      - '*.md'
+      - '.github/workflows/*.yml'
+      # re-include current file to not be excluded
+      - '!.github/workflows/build-emsdk.yml'
 
 jobs:
   build:
     runs-on: ubuntu-latest
+    env:
+      # pin SDK version to the latest, update manually
+      SDK_VERSION: 0.0.1
+      SDK_ARCHIVE: python-wasm-sdk-stable.tar.bz2
+
+      # use the most recent cython from github, but pin on a commit for CI
+      # stability. This is also needed to benefit from caching cython installs
+      LATEST_CYTHON_COMMIT: 6ac2422b48b689b021a48dff9ee14095232baafe
+
     steps:
-    - uses: actions/checkout@v1
-    - name: pygame-wasm-builder prepare
+    - uses: actions/checkout@v3.0.2
+    
+    - name: Cache pip directory for cython
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache/pip
+        key: wasm-ubuntu-cython-${{ env.LATEST_CYTHON_COMMIT }}
+
+    - name: Install latest cython and regen
       run: |
-            WD=$(pwd)
-            python3 -V
-            echo $WD
-            clang --version | head -n 1
-            echo
-            grep "^Pkg.Revision =" ${ANDROID_HOME}/ndk-bundle/source.properties
-            sudo apt-get update
-            sudo apt-get install bash wget
-            cd $GITHUB_WORKSPACE/..
-            git clone https://github.com/pmp-p/python-wasm-plus.git
-            mkdir $GITHUB_WORKSPACE/../python-wasm-plus/src
-            ln -s $(pwd)/pygame $GITHUB_WORKSPACE/../python-wasm-plus/src/pygame-wasm
-            cd python-wasm-plus
-            bash ./python-wasm-plus.sh
-            bash ./buildapp.sh
+        pip install git+https://github.com/cython/cython.git@${{ env.LATEST_CYTHON_COMMIT }}
+        touch $(find |grep pxd$)
+        python3 setup.py cython_only
 
+    - name: Install python-wasm-sdk
+      run: |
+        mkdir python-wasm-sdk
+        cd python-wasm-sdk
+        curl -sL --retry 5 https://github.com/pygame-web/python-wasm-sdk/releases/download/$SDK_VERSION/$SDK_ARCHIVE > $SDK_ARCHIVE
+        tar -xf $SDK_ARCHIVE
+        mkdir src
+        ln -s $GITHUB_WORKSPACE src/pygame-wasm
+        ls -A
+      working-directory: ../
+
+    - name: Run config
+      run: ./config
```

Comment:
The config file still seems to be missing here somehow...

I'm not a fan of hardcoding things to specific paths, as these paths can break if at some point we need to change our CI service for any reason

Reply:
> I'm not a fan of hardcoding things to specific paths, as these paths can break if at some point we need to change our CI service for any reason
indeed, but  cpython hardcode paths for its stdlib ( both build python and wasm python ) and most important in sysconfig ( compiler directives). all prebuilt generated by emsdk hardcode the "--prefix" path in various config / pkgconfig scripts used as a sysroot.
i guess that sdk will never be relocatable.

Reply:
hmm... if we have no other way than hardcoding, I think it would be better to have the hardcoded path as something like `/home/runner/work/pygame-web/python-wasm-sdk` for consistency, and to correctly match the dir on the sdk repo CI

Reply:
Then what about /tmp/python-wasm-sdk so users can use sdk without elevated privileges to unpack the tarball on linux and osx ?

Reply:
I'm not a linux filesystem expert in any way, but I think `tmp` is the wrong directory to use here? I understand all users have permissions to put stuff here, but an sdk install is not something that should be temporary. Perhaps we should use something like `opt` although this would need elevated perms IG.

Other options include `~/.local`, or something like `~/.python-wasm-sdk` (this is a pattern that some other stuff follows)

Reply:
opt indeed is usually not writeable, neither /usr/local ,  /tmp is safe to use and is permissive for exec rights which are expected for sdk.    ~/ would not work because it is expand to $HOME that we cannot predict.

> but an sdk install is not something that should be temporary.

i agree, then the tradeoff could be   /opt/sdk/python-wasm ?

Reply:
Maybe `/opt/python-wasm-sdk`? Considering the convention for apps using `opt` is usually `/opt/$NAME`

Reply:
> Considering the convention for apps using opt is usually /opt/$NAME

ok then /opt/python-wasm-sdk it is, i'll change https://github.com/pygame-web/python-wasm-sdk CI to match that and make the tarball relative to / so there is no ambiguity to it's correct place.

Reply:
Updated this PR too, and force pushed with updated changes from main to fix conflicts